libs/build_miniconda3.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# Download and install Miniconda3
+# https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html
+
+set -eux
+
+name="miniconda3"
+version=${2:-${STACK_miniconda3_version:-"latest"}}
+pyversion=${3:-${STACK_miniconda3_pyversion:-}}
+
+if $MODULES; then
+  prefix="${PREFIX:-"/opt/modules"}/core/$name/$version"
+  if [[ -d $prefix ]]; then
+    [[ $OVERWRITE =~ [yYtT] ]] && ( echo "WARNING: $prefix EXISTS: OVERWRITING!"; $SUDO rm -rf $prefix; $SUDO mkdir $prefix ) \
+                               || ( echo "WARNING: $prefix EXISTS, SKIPPING"; exit 1 )
+  fi
+else
+  prefix=${MINICONDA3_ROOT:-"/usr/local"}
+fi
+
+cd ${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}
+
+host=$(uname -s)
+
+if [[ "$host" == "Darwin" ]]; then
+  os="MacOSX"
+elif [[ "$host" == "Linux" ]]; then
+  os="Linux"
+else
+  echo "Unsupported host $host for Miniconda3; ABORT!"
+  exit 1
+fi
+
+software=$name-$version
+pkg_version=$version
+[[ -n ${pyversion:-} ]] && pkg_version=$pyversion_$version
+installer="Miniconda3-${pkg_version}-${os}-x86_64.sh"
+URL="https://repo.anaconda.com/miniconda/$installer"
+
+[[ -d $software ]] || ( mkdir -p $software; $WGET $URL -O $software/$installer )
+[[ ${DOWNLOAD_ONLY} =~ [yYtT] ]] && exit 0
+
+$SUDO bash $software/$installer -b -p $prefix -s
+set +x
+echo "sourcing conda.sh"
+source $prefix/etc/profile.d/conda.sh
```

Comment:
I get an unbounb variable `PS1` when sourcing this. Should add a `set +eu` around it.

```
set +eu
source $prefix/etc/profile.d/conda.sh
set -eu
```

Reply:
> I get an unbounb variable `PS1` when sourcing this. Should add a `set +eu` around it.
> 
> ```
> set +eu
> source $prefix/etc/profile.d/conda.sh
> set -eu
> ```

This is an known bug in the conda scripts, I think. If your suggestion works, then great.

Reply:
That didn't work, but `export PS1=""` worked. Not sure if there's a better way.

Reply:
instead of compromising on error trapping, what if we do:
```
PS1=
source $prefix/etc/profile.d/conda.sh
```

Reply:
Yes, that works.

Reply:
```suggestion
PS1=
source $prefix/etc/profile.d/conda.sh
```
@kgerheiser Can I commit this change?

Reply:
Yes